### PR TITLE
Feature/CORE-538/CORE-575/prepare e2e testing dependencies

### PR DESCRIPTION
**Pull request by safydy-r**
[overview of the 1st test]
core-575

### CODE REVIEW

**Review by object-required — APPROVED** [review]
This looks amazing :fire:

File: cypress/support/commands.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/// <reference types="cypress" />
```
Comment (safydy-r) [inline cypress/support/commands.ts:1]: auto generated by cypress as examples

File: tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "esnext",
+    "target": "ES2018",
```
Comment (safydy-r) [inline tsconfig.json:3]: this is important to allow cypress understanding the `??` & `?.` operators

File: src/views/NewFlow.vue [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
               <a
                 :href="route('flows.editFlow', {flowId: activeFlow.uuid, component: 'builder', mode: 'edit'})"
-                class="btn btn-primary"
+                class="btn btn-primary cy--create--btn"
```
Comment (bulatgab) [inline src/views/NewFlow.vue:29]: what do you think about using `data-` attributes in the future and maybe even custom commands like `getBySel`? (see here: https://docs.cypress.io/guides/references/best-practices#Selecting-Elements)

File: src/main.ts [hunk 4]
```diff
@@ -51,6 +61,14 @@ async function main(): Promise<void> {
     provide: {store},
     render: (h) => h(Vue.extend(App)),
   }).$mount('#app')
+
+  if (window?.Cypress !== undefined) {
+    console.debug('Cypress detected, exposing app on window')
+    // This will allow us to test vuex store in Cypress
+    window.store = store
```
Comment (bulatgab) [inline src/main.ts:68]: I'm not a fan of global objects but I couldn't find a better approach 👍

**Review by bulatgab — APPROVED** [review]
So cool to see e2e tests 👍
Thanks!

File: src/main.ts (hunk 4 above)
Comment (safydy-r) [reply src/main.ts:68]: Same here, I look for alternative but couldn't find a working one. 
That's why I put it under the condition `window?.Cypress !== undefined` to not set that global var on prod

File: src/views/NewFlow.vue (hunk 3 above)
Comment (safydy-r) [reply src/views/NewFlow.vue:29]: Good point, I forgot to check to recommendation. I think I'm gonna make changes at this PR. Thanks @bulatgab

File: cypress/e2e/undo-redo/create-new-flow.cy.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    const getStore = () => cy.window().its('store')
+    const flowModuleState = () => getStore().its('state.flow')
+    const flowsListState = () => flowModuleState().its('flows')
+    flowsListState().should('have.length', 0)
+    cy.get('[data-cy="create-flow--btn"]').click()
+
+    // ############ Create flow page ############
+    const flowName = 'test flow'
+    const interactionTimeout = 1000
+    cy.get('[data-cy="create--btn"]').click()
```
[the remark this answers is not in the document]
Comment (bulatgab) [reply cypress/e2e/undo-redo/create-new-flow.cy.js:18]: Thanks :)

**Pull request merged**